### PR TITLE
Improve generated code detection to look for <auto-generated> comments at top of file

### LINKED ISSUE #16: CSE0003	Consider using an expression-bodied member in _Layout.cshtml line 1
In fact I get this in all cshtml files.

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #16

### CODE REVIEW

File: Source/CSharpEssentials/Extensions.cs [hunk 1]
```diff
@@ -168,9 +168,83 @@ private static bool IsGeneratedCode(string filePath)
             return false;
         }
 
-        public static bool IsGeneratedCode(this SyntaxTree tree)
+        private static bool Matches(string s, string value, int startIndex)
         {
-            return IsGeneratedCode(tree.FilePath);
+            if (startIndex + value.Length > s.Length)
```
Comment (pharring) [inline Source/CSharpEssentials/Extensions.cs]: ```string.Compare(...)```

**Comment by pharring** [comment]
:+1:

File: Source/CSharpEssentials/Extensions.cs (hunk 1 above)
Comment (DustinCampbell) [inline Source/CSharpEssentials/Extensions.cs]: Thanks!

**Comment by DustinCampbell** [comment]
@CyrusNajmabadi might also be interested in reviewing

File: Source/CSharpEssentials/Extensions.cs [hunk 2]
```diff
@@ -168,9 +168,68 @@ private static bool IsGeneratedCode(string filePath)
...
+
+                    // Scan past whitespace.
+                    int index = 2;
+                    while (index < text.Length && char.IsWhiteSpace(text[index]))
+                    {
+                        index++;
+                    }
+
+                    // If there's nothing else in the string, we're done.
+                    if (index == text.Length)
```
Comment (pharring) [inline Source/CSharpEssentials/Extensions.cs]: This check is redundant because string.Compare will do it too.

Comment (DustinCampbell) [inline Source/CSharpEssentials/Extensions.cs]: Sigh... thanks Paul. :disappointed:

**Comment by pharring** [comment]
:+1:

**Pull request merged**